datalad/distribution/dataset.py

```diff
@@ -126,11 +128,17 @@ def repo(self):
                             lgr.debug("Detected %s at %s", cls, self._path)
                             self._repo = cls(self._path, create=False, **kw)
                             break
-                        except (InvalidGitRepositoryError, NoSuchPathError, RuntimeError) as exc:
-                            lgr.debug("Oops -- guess on repo type was wrong?: %s", exc_str(exc))
+                        except (InvalidGitRepositoryError, NoSuchPathError) as exc:
+                            lgr.debug(
+                                "Oops -- guess on repo type was wrong?: %s",
+                                exc_str(exc))
                             pass
-                if self._repo is None:
-                    lgr.info("Failed to detect a valid repo at %s" % self.path)
+                        # version problems come as RuntimeError: DO NOT CATCH!
+            if self._repo is None:
+                # Often .repo is requested to 'sense' if anything is installed
```

Comment:
actually started to work on that but decided that it better goes after as a separate PR -- otherwise would be lost here and left without due review